server/bin/gold/test-5.2.txt

```diff
@@ -118,6 +118,36 @@ len(actions) = 1
     }
 ]
 --- Finished pbench-unpack-tarballs (status=0)
++++ Running pbench-unpack-tarballs none re-unpack
+Template:  pbench-unittests.v3.server-reports
+Index:  pbench-unittests.v3.server-reports.1970-01 1
+len(actions) = 1
+[
+    {
+        "_id": "5ca1ab1e70015f100dedfab1ed0ff1ce",
+        "_index": "pbench-unittests.v3.server-reports.1970-01",
+        "_op_type": "create",
+        "_source": {
+            "@generated-by": {
+                "commit_id": "unit-test",
+                "group_id": 43,
+                "hostname": "example.com",
+                "pid": 42,
+                "user_id": 44,
+                "version": ""
+            },
+            "@timestamp": "1970-01-01T00:00:42",
+            "chunk_id": 1,
+            "doctype": "status",
+            "name": "pbench-re-unpack-tarballs",
+            "text": "pbench-re-unpack-tarballs.run-1970-01-01T00:00:42-UTC(unit-test) - w/ 0 errors\nProcessed 0 result tar balls, 0 successfully, 0 warnings, 0 errors, and 0 duplicates\n\n",
```

Comment:
I think this is the re-unpack report: nothing was restored so nothing was processed. The previous scenario involved unpacking "new" tar balls.

This was an existing test which (because of a config file change) is now running one extra cron job: the cron job produced some output (so a change to the gold file was needed) but it didn't do anything (because there was nothing for it to do).

Reply:
Nick, that all makes good sense in context, but I still have to ask, is this what was _supposed_ to have happened?  Is this this test that we _wanted_ it to do?

Are we really going to run a periodic cron job which, once we've recovered from the power shutdown, will never actually do anything (until the next time someone wants to re-unpack a tarball)?  Seems weird.

Reply:
I think the ratinonale (such as it is) is that  the (legacy) unittests script is supposed to test all of the cron jobs that could potentially be enabled on a real server (about a dozen `_run_allscripts` tests do that - the rest are more specific). Some tests override the  global `state/config/pbench-server.cfg` with their own, but none of them override the cron settings, so those are inherited from the "real" `pbench-server-default.cfg` file. So that set of tests runs all of the cron job scripts, including new ones that are added to the `pbench-server-default.cfg` file: I think that's a good thing. Of course, targeted tests for the new functionality have to be added as well, but at least we know (for some version of "know") that we are not going to keel over and die immediately if the script is added.

As for the second part, I think the ultimate answer is that we not going to run *any* cron jobs: the service that will replace all this is going to figure out what needs to be done (based on some currently unspecified input) and dispatch the appropriate service module instantly. If nothing is ever restored from backup, then the service module will remain unused, but even after we've recovered from the current predicament, a restore might not be such an uncommon occurrence.

Reply:
> Nick, that all makes good sense in context, but I still have to ask, is this what was _supposed_ to have happened? Is this the test that we _wanted_ it to do?

Yes.

> Are we really going to run a periodic cron job which, once we've recovered from the power shutdown, will never actually do anything (until the next time someone wants to re-unpack a tarball)? Seems weird.

No.  We don't have the "re-unpack" cronjob enabled by default.  So one would have to update the configuration file to add that step, and then rebuild the cronjob file to add it.